plugins/module_utils/data_set.py

```diff
@@ -203,6 +203,15 @@ def ensure_present(
         changed = False
         if DataSet.data_set_cataloged(name, tmphlq=tmp_hlq):
             present = True
+        if present and not replace and volumes:
+            cataloged_volumes = DataSet.data_set_cataloged_volume_list(name, tmphlq=tmp_hlq)
+            requested_volumes = [v.upper() for v in volumes]
```

Comment:
Just suggest we do it the same way as we did below.
```suggestion
            requested_volumes = [vol.upper() for vol in volumes]
```

Reply:
resolved